cluster/gce/trusty/configure.sh

```diff
@@ -22,7 +22,8 @@
 
 config_hostname() {
   # Set the hostname to the short version.
-  short_hostname=$(hostname -s)
+  host_name=$(curl --silent "http://metadata.google.internal/computeMetadata/v1/instance/hostname" -H "Metadata-Flavor: Google")
```

Comment:
Here's how we do it (in go code): https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/gce/gce.go#L121-L131